### ISSUE TITLE
API for controlling playback, authentication... 

### ISSUE THREAD

**Issue by tommyzat** [librespot-org/librespot#1116]
1. Is it possible to have an API for controlling playback, kicking users even?

2. How would I go about filtering who can and can't connect to librespot?

**Comment by kingosticks** [comment]
Yes it is possible, but we only currently implement a Spotify Connect receiver i.e. for playback. We don't implement a (remote) controller side. Related projects like [spotcontrol](https://github.com/badfortrains/spotcontrol) do implement this. 

Alternatively, users can control playback via [Spotify's Player Web API](https://developer.spotify.com/documentation/web-api/reference/#/operations/start-a-users-playback). Librespot can [provide an access token](https://github.com/librespot-org/librespot/blob/dev/examples/get_token.rs#L25), or you can obtain one yourself using Spotify's API endpoints. I guess transferring your playback session to the device will kick off anyone already using it?

Librespot does provide a [Disconnect](https://github.com/librespot-org/librespot/blob/3662302196c1aeb35501db717b88152c50212117/connect/src/spirc.rs#L438) method but you'd need to write additional code to expose it. You'd also need to write additional code to implement access control/user filtering, nothing like that exists. I don't think we have any extensions like that. If it did exist, it'd probably belong in `ConnectConfig`.

**Comment by tommyzat** [comment]
Thank you. I think a DBus interface would prove to be useful! I'm not really sure how I'd expose those functions

**Comment by roderickvd** [comment]
spotifyd, a related project, exposes an MPRIS interface.
